### PR TITLE
Krypton updates

### PR DESCRIPTION
**Pull request by phil65**

### CODE REVIEW

File: script.extendedinfo/resources/lib/RottenTomatoes.py [hunk 1]
```diff
@@ -3,19 +3,23 @@
...
 
 
 def get_movies(movie_type):
-    movies = []
+    movies = ItemList(content_type="movies")
     url = '%s.json?apikey=%s' % (movie_type, RT_KEY)
-    results = Utils.get_JSON_response(BASE_URL + url, folder="RottenTomatoes")
+    results = utils.get_JSON_response(BASE_URL + url, folder="RottenTomatoes")
+    if "error" in results:
+        utils.notify("Error", results["error"])
```
Comment (learningit) [inline script.extendedinfo/resources/lib/RottenTomatoes.py]: Should this string be in strings.po?

File: script.extendedinfo/resources/lib/TheMovieDB.py [hunk 2]
```diff
@@ -341,12 +283,12 @@ def handle_movies(results, local_first=True, sortkey="year"):
...
                         'genre': " / ".join([i for i in genres if i]),
                         'votes': movie.get('vote_count'),
-                        'year': Utils.get_year(release_date),
+                        'year': utils.get_year(release_date),
                         'rating': round(movie['vote_average'], 1) if movie.get('vote_average') else "",
                         'userrating': movie.get('rating'),
                         'premiered': release_date})
         item.set_properties({'id': movie.get("id"),
-                             'popularity': movie.get('popularity'),
+                             'popularity': round(movie['popularity'], 1) if movie.get('popularity') else "",
```
Comment (learningit) [inline script.extendedinfo/resources/lib/TheMovieDB.py]: Do you really want to do this? I see a float or empty string assignment here. Will cause an error downstream if someone tries to do anything like int(), won't it? Same thing line 287.

File: script.extendedinfo/resources/lib/RottenTomatoes.py (hunk 1 above)
Comment (phil65) [inline script.extendedinfo/resources/lib/RottenTomatoes.py]: yup, could get moved. Will do that for next update, thx.

File: script.extendedinfo/resources/lib/TheMovieDB.py (hunk 2 above)
Comment (phil65) [inline script.extendedinfo/resources/lib/TheMovieDB.py]: It's fine this way, I filter out all empty dict entries before processing any further.

File: script.extendedinfo/resources/lib/TheMovieDB.py [hunk 3]
```diff
@@ -693,75 +703,80 @@ def get_show_tmdb_id(tvdb_id=None, source="tvdb_id"):
...
+    return response["tv_results"][0]["id"]
+
+
+def get_show_id(tmdb_id=None, return_id="imdb_id"):
+    params = {"append_to_response": "external_ids",
+              "language": addon.setting("LanguageID")}
+    response = get_data(url="tv/%s" % (tmdb_id),
+                        params=params)
+    if not response:
+        utils.notify("TVShow Info not available.")
```
Comment (learningit) [inline script.extendedinfo/resources/lib/TheMovieDB.py]: Assuming that this is a notification, shouldn't this be in strings.po?

File: script.extendedinfo/resources/lib/TheMovieDB.py [hunk 4]
```diff
@@ -693,75 +703,80 @@ def get_show_tmdb_id(tvdb_id=None, source="tvdb_id"):
...
 
 
 def get_trailer(movie_id):
-    response = get_full_movie(movie_id)
+    response = get_movie(movie_id)
     if response and "videos" in response and response['videos']['results']:
         return response['videos']['results'][0]['key']
-    Utils.notify("Could not get trailer")
-    return ""
+    utils.notify("Could not get trailer")
```
Comment (learningit) [inline script.extendedinfo/resources/lib/TheMovieDB.py]: Same here.

File: script.extendedinfo/resources/lib/TheMovieDB.py (hunk 3 above)
Comment (phil65) [inline script.extendedinfo/resources/lib/TheMovieDB.py]: yup, will keep in mind for next update.

File: script.extendedinfo/resources/lib/TheMovieDB.py [hunk 5]
```diff
@@ -693,75 +703,80 @@ def get_show_tmdb_id(tvdb_id=None, source="tvdb_id"):
...
-              "include_image_language": "en,null,%s" % addon.setting("LanguageID")}
-    if Login.check_login():
-        params["session_id"] = Login.get_session_id()
-    info = get_data(url="movie/%s" % (movie_id),
-                        params=params,
-                        cache_days=cache_time)
+    info = get_movie(movie_id=movie_id, cache_days=cache_days)
     if not info:
-        Utils.notify("Could not get movie information")
+        utils.notify("Could not get movie information")
```
Comment (learningit) [inline script.extendedinfo/resources/lib/TheMovieDB.py]: and here?

File: script.extendedinfo/resources/lib/TheMovieDB.py (hunk 2 above)
Comment (phil65) [inline script.extendedinfo/resources/lib/TheMovieDB.py]: and sorry @Do, that was probably a typo :D

File: script.extendedinfo/resources/lib/TheMovieDB.py [hunk 6]
```diff
@@ -923,134 +941,154 @@ def extended_episode_info(tvshow_id, season, episode, cache_time=7):
...
-                         'gender': genders.get(response['gender']),
-                         'cast_id': response.get('cast_id'),
-                         'credit_id': response.get('credit_id'),
-                         'deathday': response.get('deathday'),
-                         'placeofbirth': response.get('place_of_birth'),
-                         'homepage': response.get('homepage'),
-                         "DBMovies": len([d for d in listitems["movie_roles"] if "dbid" in d])})
-    info.set_artwork(get_image_urls(profile=response.get("profile_path")))
-    return (info, listitems)
+    lists = {"movie_roles": handle_movies(data["movie_credits"]["cast"]).reduce("character"),
```
Comment (learningit) [inline script.extendedinfo/resources/lib/TheMovieDB.py]: I had to read up on this to understand how this works. Interesting, learned something new. thanks

**Comment by learningit** [comment]
Should this script.extendedinfo/script.extendedinfo.sublime-workspace be in the commit?

**Comment by phil65** [comment]
ooops, no, that was a mistake. will remove those files, thx

**Comment by phil65** [comment]
Updated.

**Pull request merged**